picard/browser/browser.py

```diff
@@ -38,18 +39,52 @@
 from picard.util import mbid_validate
 
 
-def response(code):
+def response(code, origin=None):
     if code == 200:
         resp = '200 OK'
     elif code == 400:
         resp = '400 Bad Request'
     else:
         resp = '500 Internal Server Error'
+    if origin:
+        allowOrigin = 'Access-Control-Allow-Origin: {}\r\n'.format(origin)
+    else:
+        allowOrigin = ''
     return bytearray(
         'HTTP/1.1 {}\r\n'
         'Cache-Control: max-age=0\r\n'
+        '{}'
         '\r\n'
-        'Nothing to see here.\r\n'.format(resp), 'ascii')
+        'Nothing to see here.\r\n'.format(resp, allowOrigin), 'ascii')
+
+
+def _read_origin(conn):
+    while True:
+        line = conn.readLine().data()
+        if not line or line == b'\r\n':
+            break
+        if line.startswith(b'Origin:'):
```

Comment:
Since that's an http header, the match is too strict.

https://tools.ietf.org/html/rfc7230#section-3.2

> Each header field consists of a case-insensitive field name followed
>    by a colon (":"), optional leading whitespace, the field value, and
>    optional trailing whitespace.

Perhaps we should use some already existing functions for that, perhaps https://docs.python.org/3/library/http.client.html#http.client.parse_headers (dunno if appropriate in this context)